### PR TITLE
Bundler: Align error messages in `queryRubygems`

### PR DESCRIPTION
**Pull request by mnonnenmacher**
Make sure all error messages mention the gem meta-data was requested
for, and add a specific error message for bad gateway.

---
This change is [Reviewable](https://reviewable.io/reviews/heremaps/oss-review-toolkit/1612)

### CODE REVIEW

File: analyzer/src/main/kotlin/managers/Bundler.kt [hunk 1]
```diff
@@ -253,18 +253,22 @@ class Bundler(
 
                 OkHttpClientHelper.HTTP_TOO_MANY_REQUESTS -> {
                     throw IOException(
-                        "RubyGems reported too many requests, see " +
+                        "RubyGems reported too many requests when requesting meta-data for gem '$name', see " +
```
Comment (sschuberth) [inline analyzer/src/main/kotlin/managers/Bundler.kt]: Commit message nit: Drop "was".

File: analyzer/src/main/kotlin/managers/Bundler.kt [hunk 2]
```diff
@@ -253,18 +253,22 @@ class Bundler(
...
-                else -> {
-                    if (retryCount > 0 && response.code == HttpURLConnection.HTTP_BAD_GATEWAY) {
+                HttpURLConnection.HTTP_BAD_GATEWAY -> {
+                    if (retryCount > 0) {
                         // We see a lot of sporadic "bad gateway" responses that disappear when trying again.
                         Thread.sleep(100)
                         return queryRubygems(name, version, retryCount - 1)
                     }
 
+                    throw IOException("RubyGems reported bad gateway when requesting meta-data for gem '$name'.")
```
Comment (sschuberth) [inline analyzer/src/main/kotlin/managers/Bundler.kt]: Maybe say "too many bad gateway errors"?

File: analyzer/src/main/kotlin/managers/Bundler.kt (hunk 1 above)
Comment (mnonnenmacher) [reply analyzer/src/main/kotlin/managers/Bundler.kt]: Done, rephrased the sentence to make it easier to understand.

File: analyzer/src/main/kotlin/managers/Bundler.kt (hunk 2 above)
Comment (mnonnenmacher) [reply analyzer/src/main/kotlin/managers/Bundler.kt]: Done.

**Pull request merged**